Providers/RustSyncManager.swift

```diff
@@ -0,0 +1,635 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/
+
+import Account
+import Shared
+import Storage
+import Sync
+import SyncTelemetry
+import AuthenticationServices
+import Common
+
+// Extends NSObject so we can use timers.
+public class RustSyncManager: NSObject, SyncManager {
+    // We shouldn't live beyond our containing BrowserProfile, either in the main app
+    // or in an extension.
+    // But it's possible that we'll finish a side-effect sync after we've ditched the
+    // profile as a whole, so we hold on to our Prefs, potentially for a little while
+    // longer. This is safe as a strong reference, because there's no cycle.
+    unowned fileprivate let profile: BrowserProfile
+    fileprivate let prefs: Prefs
+    fileprivate var constellationStateUpdate: Any?
+    fileprivate var syncTimer: Timer?
+    fileprivate var backgrounded: Bool = true
+    private let logger: Logger
+
+    let FifteenMinutes = TimeInterval(60 * 15)
+
+    deinit {
+        if let c = constellationStateUpdate {
+            NotificationCenter.default.removeObserver(c)
+        }
```

Comment:
nit: avoid single letter var
```suggestion
        if let stateUpdate = constellationStateUpdate {
            NotificationCenter.default.removeObserver(stateUpdate)
        }
```